tests/functional_tests/ffront_tests/test_math_builtin_execution.py

```diff
@@ -0,0 +1,118 @@
+import numbers
+from typing import Callable
+
+import numpy as np
+import pytest
+
+from functional.fencil_processors import roundtrip
+from functional.ffront import (
+    common_types as ct,
+    fbuiltins,
+    field_operator_ast as foast,
+    symbol_makers,
+)
+from functional.ffront.decorator import FieldOperator
+from functional.ffront.fbuiltins import Dimension, Field, float64, int32, int64
+from functional.ffront.foast_passes.type_deduction import FieldOperatorTypeDeduction
+from functional.ffront.source_utils import CapturedVars
+from functional.iterator.embedded import np_as_located_field
+
+from ..iterator_tests.math_builtin_test_data import math_builtin_test_data
+
+
+fieldview_backend = roundtrip.executor
+
+IDim = Dimension("IDim")
+
+# TODO(tehrengruber): add tests for scalar arguments to builtin. To avoid code
+#  bloat this is postponed until programatically creating field operators
+#  becomes easier.
+
+
+def make_builtin_field_operator(builtin_name: str):
+    # TODO(tehrengruber): creating a field operator programmatically should be
+    #  easier than what we need to do here.
+    # construct annotation dictionary containing the input argument and return
+    #  types
+    if builtin_name in fbuiltins.UNARY_MATH_NUMBER_BUILTIN_NAMES:
+        annotations = {"inp": Field[[IDim], float64], "return": Field[[IDim], float64]}
+    elif builtin_name in fbuiltins.UNARY_MATH_FP_BUILTIN_NAMES:
+        annotations = {"inp": Field[[IDim], float64], "return": Field[[IDim], float64]}
+    elif builtin_name in fbuiltins.UNARY_MATH_FP_PREDICATE_BUILTIN_NAMES:
+        annotations = {"inp": Field[[IDim], float64], "return": Field[[IDim], bool]}
+    elif builtin_name in fbuiltins.BINARY_MATH_NUMBER_BUILTIN_NAMES:
+        annotations = {
+            "inp1": Field[[IDim], float64],
+            "inp2": Field[[IDim], float64],
+            "return": Field[[IDim], float64],
+        }
+    else:
+        raise AssertionError(f"Unknown builtin `{builtin_name}`")
+
+    captured_vars = CapturedVars(
+        nonlocals={"IDim": IDim},
+        globals={builtin_name: getattr(fbuiltins, builtin_name)},
+        annotations=annotations,
+        builtins=set(),
+        unbound=set(),
+    )
+
+    loc = foast.SourceLocation(line=1, column=1, source="none")
```

Comment:
Question: There is no easier way to fix source locations? Similar to `ast.fix_missing_locations`?

Reply:
No, this is really annoying if one generates foast expressions programmatically. Locations are mandatory to avoid accidentally not adding them in the Python -> FOAST lowering (a problem that made errors in the cartesian toolchain aweful), but this is the downside. Some way to disable this would be nice, e.g. a wrapper that behaves like the `foast` module, but automatically adds the locations whenever a node is created.